buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy

```diff
@@ -383,20 +383,43 @@ class VagrantTestPlugin implements Plugin<Project> {
         assert project.tasks.packagingTest != null
         Task packagingTest = project.tasks.packagingTest
 
+        assert project.tasks.clean != null
+        Task clean = project.tasks.clean
+
         /*
          * We always use the main project.rootDir as Vagrant's current working directory (VAGRANT_CWD)
          * so that boxes are not duplicated for every Gradle project that use this VagrantTestPlugin.
          */
+        File vagrantFile = new File(project.rootDir, 'Vagrantfile')
         def vagrantEnvVars = [
                 'VAGRANT_CWD'           : "${project.rootDir.absolutePath}",
-                'VAGRANT_VAGRANTFILE'   : 'Vagrantfile',
+                'VAGRANT_VAGRANTFILE'   : vagrantFile.name,
                 'VAGRANT_PROJECT_DIR'   : "${project.projectDir.absolutePath}"
         ]
 
+        File vagrantMarkersDir = new File(project.rootProject.buildDir, "vagrant")
+        clean.delete vagrantMarkersDir
+
         // Each box gets it own set of tasks
         for (String box : BOXES) {
             String boxTask = box.capitalize().replace('-', '')
 
+            // Destroy the box if the Vagrantfile changed.
+            Task destroy = project.tasks.create("vagrant${boxTask}#destroy", VagrantCommandTask) {
+                boxName box
+                environmentVars vagrantEnvVars
+                args 'destroy', box, '--force'
+                dependsOn vagrantCheckVersion, virtualboxCheckVersion
+            }
+            destroy.inputs.file(vagrantFile)
```

Comment:
Sorry I could have sworn the marker file was the input and output when I looked at this.